app/assets/javascripts/controllers/FileCtrl.coffee

```diff
@@ -43,37 +44,71 @@
     return
 
   ###*
-   * save file as bruse_file and append bruse_file to the file
-   * @param {obj} file  the file
+   * save file to temp array
+   * @param {obj} file  the file to add
   ###
   $scope.add = (file) ->
+    if file.is_dir and !file.contents
```

Comment:
what does all these statements represent? If we are in the right directory and the file does not exist here? or if we are in the right directory or if the file is not added? I'm just wondering because I am confused